packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.ts

```diff
@@ -220,39 +225,61 @@ export class GraphQLSitemapService {
     languages: string[],
     formatStaticPath: (path: string[], language: string) => StaticPath
   ): Promise<StaticPath[]> {
+    let paths = new Array<StaticPath>();
     if (!languages.length) {
       throw new RangeError(languageError);
     }
-    // Fetch paths using all locales
-    const paths = await Promise.all(
-      languages.map((language) => {
-        if (language === '') {
-          throw new RangeError(languageEmptyError);
-        }
-        debug.sitemap('fetching sitemap data for %s', language);
-        return this.fetchLanguageSitePaths(language).then((results) =>
-          this.transformLanguageSitePaths(results, formatStaticPath, language)
-        );
-      })
-    );
+    // Get all sites
+    const sites = this.options.sites;
+    let aggregatedPaths: StaticPath[] = [];
+
+    // Fetch paths for each site
+    for (let i = 0; i < sites.length; i++) {
+      const siteName = sites[i];
+      const multiSiteName = sites.length > 1 ? siteName : undefined;
+      const multipleLanguages = languages.length > 1;
+
+      // Fetch paths using all locales
+      paths = ((await Promise.all(
+        languages.map(async (language) => {
+          if (language === '') {
+            throw new RangeError(languageEmptyError);
+          }
+          debug.sitemap('fetching sitemap data for %s', language, siteName);
```

Comment:
Will not be logged without additional token
```suggestion
          debug.sitemap('fetching sitemap data for %s %s', language, siteName);
```